### DISCUSSION TITLE
I can't figure out how to "transition leave"

### DISCUSSION THREAD

**Discussion by joseflekardal** [nuxt/nuxt#20149] [Questions]
I do have a list route `/items` that displays a list of nuxt links like `/items/[id]`, when i click one a detail view sidebar slides in from the right. Inside the sidebar I've got a close button (nuxt-link that goes to `/items`). It is working but I can't get the slide animation to run when i click on close in my sidebar. The only time the leave transition runs is when I switch between detail views but not when closing the sidebar.

I've tried putting the definePageMeta on the parent page aswell, which also works but still no leave transition.

I've also done some logging using the onEnter, onLeave, onBeforeLeave hooks on the pageTransition property and I can confirm leave never runs when closing the sidebar.

```
$ tree -L 2
├── README.md
├── nuxt.config.ts
├── package-lock.json
├── package.json
├── pages
│   ├── items
│   └── items.vue
├── public
│   └── favicon.ico
└── tsconfig.json
```

```vue
<!-- pages/items.vue -->
<template>
  <div>
    <header>
      <h1>Page transition example</h1>
    </header>
    <main>
      <ul>
        <li><nuxt-link href="/items/1">Item 1</nuxt-link></li>
        <li><nuxt-link href="/items/2">Item 2</nuxt-link></li>
      </ul>
    </main>
    <nuxt-page />
  </div>
</template>

<style>
main,
header {
  padding: 1rem;
}

.slide-enter-active,
.slide-leave-active {
  transition: all 0.5s;
}

.slide-enter-from,
.slide-leave-to {
  transform: translateX(100%);
}
</style>

```

```vue
<!-- pages/items/[itemId].vue -->
<script setup lang="ts">
definePageMeta({
  pageTransition: {
    name: "slide",
  },
});

// i hate this but my page transition wont run without this or some other async code
await nextTick();
</script>

<template>
  <aside>
    <h2>Details for item: {{ $route.params.itemId }}</h2>
    <nuxt-link href="/items">Close</nuxt-link>
  </aside>
</template>

<style>
aside {
  position: fixed;
  right: 0;
  top: 0;
  height: 100vh;
  background: gray;
  padding: 1rem;
  width: 300px;
}
</style>
```

please tell me I stupid and that there is a simple answer to this 🙏

**Comment by joseflekardal** [comment]
I just found this issue that seems related. #12361

**Comment by smwbtech** [comment]
I faced with the same problem. Nested page element has incorrect transitions. I'm using definePageMeta method too for transition name setup.
